components/server/src/projects/projects-service.ts

```diff
@@ -204,7 +217,21 @@ export class ProjectsService {
     }
 
     async deleteProject(projectId: string): Promise<void> {
-        return this.projectDB.markDeleted(projectId);
+        try {
+            await this.projectDB.transaction(async (db) => {
+                // TODO(gpl): This is a case where we'd need to extend the service + API to also accept the orgId as first parameter
+                const project = await db.findProjectById(projectId);
+                if (!project) {
+                    throw new Error("Project does not exist");
+                }
+                await db.markDeleted(projectId);
+
+                await this.authorizer.writeRelationships(removeProjectFromOrg(project.teamId, projectId));
+            });
+        } catch (err) {
+            // As the remove relationships is the last call that might fail, we don't have to do anything here.
```

Comment:
Is this true?

It could still be the case that
1. All inside `transaction(..)` succeeds
2. But the `transaction.commit()` fails
3. Now we have removed the relationship, but we still have the project

Reply:
You are absolutely correct. Will create a follow-up. :running_man:

Reply:
Follow-up here: https://github.com/gitpod-io/gitpod/pull/18055